hack/update-image-tags.sh

```diff
@@ -4,14 +4,65 @@ set -o nounset
 set -o errexit
 set -o pipefail
 
-# update the nvidia k8s device plugin
+ECR_PUBLIC_REGISTRY="public.ecr.aws"
+EKS_CONTAINER_REGISTRY="602401143452.dkr.ecr.us-west-2.amazonaws.com"
+
+# get_ecr_image_tags <REGISTRY> <REPOSITORY>
+# e.g. get_ecr_image_tags $ECR_PUBLIC_REGISTRY amazonlinux/amazonlinux
+get_ecr_image_tags() {
+    local REGISTRY=$1 
+    local REPOSITORY=$2
+
+    # Get ECR public token if image is from a public registry, otherwise use a private token
+    # An authorization token is required for every ECR HTTP request
+    if [ "$REGISTRY" = "$ECR_PUBLIC_REGISTRY" ]; then
+        local TOKEN=$(aws ecr-public get-authorization-token --region us-east-1 --output=text --query 'authorizationData.authorizationToken' 2>/dev/null)
+        local AUTHORIZATION_TYPE="Bearer"
+    else 
+        local TOKEN=$(aws ecr get-authorization-token --output text --query 'authorizationData[].authorizationToken' 2>/dev/null)
+        local AUTHORIZATION_TYPE="Basic"
+    fi
+
 
+    if [ -n "${TOKEN}" ]; then
+        curl -s -H "Authorization: ${AUTHORIZATION_TYPE} $TOKEN" "https://$REGISTRY/v2/$REPOSITORY/tags/list" | jq '.tags'
+    else 
+        echo ""
+    fi
+}
+
+# update_image_uris REPOSITORY IMAGE_TAG
+update_image_uris() {
+    local REPOSITORY=$1
+    local NEW_TAG=$2
+    PREFIX="image: ${REPOSITORY}"
+    find ./test/manifests -type f -exec sed -i "s#$PREFIX:.*#$PREFIX:$NEW_TAG#g" {} +
+}
+
+# update the nvidia k8s device plugin
+echo "Updating Nvidia device plugin image"
 NVIDIA_DEVICE_PLUGIN=$(curl -s 'https://catalog.ngc.nvidia.com/api/containers/images?orgName=nvidia&name=k8s-device-plugin&isPublic=true' | jq -r '.images | sort_by(.updatedDate) | reverse | map(select(.tag | test("^v[0-9]+.[0-9]+.[0-9]+$"))) | first | .tag')
 PREFIX="image: nvcr.io/nvidia/k8s-device-plugin"
 find ./test/manifests -type f -exec sed -i "s#$PREFIX:.*#$PREFIX:$NVIDIA_DEVICE_PLUGIN#g" {} +
```

Comment:
update this code too?
```suggestion
NVIDIA_DEVICE_PLUGIN_TAG=$(curl -s 'https://catalog.ngc.nvidia.com/api/containers/images?orgName=nvidia&name=k8s-device-plugin&isPublic=true' | jq -r '.images | sort_by(.updatedDate) | reverse | map(select(.tag | test("^v[0-9]+.[0-9]+.[0-9]+$"))) | first | .tag')
update_image_uris nvcr.io/nvidia/k8s-device-plugin $NVIDIA_DEVICE_PLUGIN_TAG
```